R/import_mzroll.R

```diff
@@ -168,6 +168,110 @@ process_mzroll <- function(mzroll_db_path,
   return(mzroll_list)
 }
 
+#' Process Metabolon
+#' 
+#' Import results generated by metabolon into romic structure, using series of CSV
+#' output files delivered by Metabolon as inputs.
+#' 
+#' @param peak_areas_path path to peak areas CSV file from metabolon results
+#' @param chem_anno_path path to chemical annotations CSV file from metabolon results
+#' @param metadata_path path to metadata CSV file from metabolon results
+#' @param peak_floor_intensity value to use to replace NA peak measurements (missing values)
+#' 
+#' @export
+process_metabolon <- function(peak_areas_path,
+                              chem_anno_path,
+                              metadata_path,
+                              peak_floor_intensity = 4096
+                              ) {
+  
+  checkmate::assertFileExists(peak_areas_path)
+  checkmate::assertFileExists(chem_anno_path)
+  checkmate::assertFileExists(metadata_path)
+  checkmate::assertNumeric(peak_floor_intensity)
+  
+  # import metabolon results as CSV files
+  peaks <- read.csv(file=peak_areas_path, check.names = FALSE)
+  anno <- read.csv(file=chem_anno_path)
+  metadata <- read.csv(file=metadata_path)
+  
+  columns_to_pivot <- colnames(peaks)
+  columns_to_pivot <- columns_to_pivot[2:length(columns_to_pivot)]
+  samples <- metadata %>%
+    select(PARENT_SAMPLE_NAME,CLIENT_SAMPLE_NUMBER) %>%
+    rename(sampleName=PARENT_SAMPLE_NAME,sampleId=CLIENT_SAMPLE_NUMBER) %>%
+    mutate(sampleId = factor(sampleId, levels = sampleId))
```

Comment:
you need to include the namespace (e.g., dplyr::select) to pass R CMD check since dplyr is not imported

Reply:
(looks like the R CMD check action was not running since it was disabled due to inactivity on the repo  :( )